tests/e2e/e2e_test.go

```diff
@@ -587,3 +587,21 @@ func TestFailOnEcsUsageAsPlugin(t *testing.T) {
 		})
 	})
 }
+
+func TestComposeContainerName(t *testing.T) {
+	c := NewParallelE2eCLI(t, binDir)
+	t.Run("compose up container_name with scale > 1", func(t *testing.T) {
+		res := c.RunDockerOrExitError("-f", "../composefiles/container_name/docker-compose-bad.yml", "compose", "up", "-d")
+		res.Assert(t, icmd.Expected{
+			ExitCode: 1,
+			Err: "WARNING: The \"web\" service is using the custom container name \"myname\". " +
+				"Docker requires each container to have a unique name. " +
+				"Remove the custom name to scale the service.",
+		})
+	})
+
+	t.Run("compose up container_name with scale == 1", func(t *testing.T) {
+		c.RunDockerCmd("-f", "../composefiles/container_name/docker-compose-good.yml", "compose", "up", "-d")
+		c.RunDockerCmd("-f", "../composefiles/container_name/docker-compose-good.yml", "compose", "down")
+	})
+}
```

Comment:
Maybe rather add a unit test on ensureService() in convergence.go for the failure case, E2E tests will become slow at some point...

Reply:
PTAL